src/onnx/parse_rotary_embedding.cpp

```diff
@@ -0,0 +1,395 @@
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+struct rotary_parameters
+{
+    // Extracted from Inputs
+    std::size_t batch_size  = 0; // Batch used by input
+    std::size_t seq_len     = 0; // Sequence length used by input
+    std::size_t head_size   = 0; // Head size used for offset in each block
+    std::size_t max_seq_len = 0; // Sequence length used by sin/cos caches
+    std::size_t hidden_size = 0; // Hidden size used by the input
+    // input shape:
+    // true => [batch_size, num_heads, seq_len, head_size]
+    // false => [batch_size, seq_len, hidden_size=num_heads*head_size]
+    bool is_bnsh = false;
+
+    // Extracted from both
+    std::size_t num_heads = 0;     // num_heads = hidden_size / head_size  or input
+    bool head_diff        = false; // head_size > rotary_embedding_dim
+
+    // Extracted from Attributes
+    std::size_t rotary_embedding_dim = 0;
+    bool interleaved                 = false;
+    bool is_packed_batching          = false;
+    float scale                      = 1.0;
+};
+
+struct parse_rotary_embedding : op_parser<parse_rotary_embedding>
+{
+    std::vector<op_desc> operators() const { return {{"RotaryEmbedding"}}; }
+
+    static void parse_attributes(const onnx_parser& parser,
+                                 const onnx_parser::node_info& info,
+                                 rotary_parameters& param)
+
+    {
+        if(contains(info.attributes, "interleaved"))
+        {
+            param.interleaved = parser.parse_value(info.attributes.at("interleaved")).at<bool>();
+        }
+
+        if(contains(info.attributes, "is_packed_batching"))
+        {
+            // Ragged batching is a form of dynamic batching where inputs of varying lengths
+            // are padded and batched together. E.g. inputs of [1, 3], [1, 4], [1, 5]
+            // would each be padded to [1, 5] and batched together as [3, 5]
+            param.is_packed_batching =
+                parser.parse_value(info.attributes.at("is_packed_batching")).at<bool>();
+            if(param.is_packed_batching)
+            {
+                MIGRAPHX_THROW(
+                    "RotaryEmbedding: is_packed_batching aka ragged batching is not supported.");
+            }
+        }
+
+        if(contains(info.attributes, "num_heads"))
+        {
+            param.num_heads = parser.parse_value(info.attributes.at("num_heads")).at<std::size_t>();
+        }
+
+        if(contains(info.attributes, "rotary_embedding_dim"))
+        {
+            param.rotary_embedding_dim =
+                parser.parse_value(info.attributes.at("rotary_embedding_dim")).at<std::size_t>();
+        }
+
+        if(contains(info.attributes, "scale"))
+        {
+            // ContribOp spec does not specify how to apply the scale and ORT neither
+            // handles nor tests this attribute
+            param.scale = parser.parse_value(info.attributes.at("scale")).at<float>();
```

Comment:
May work like RoPE scaling I think. Right now default is a scale of 1 so I assume you would use the default scaling

Found this from minor looking around. 
https://github.com/huggingface/transformers/issues/24472

Something we can document and add later.